### PR TITLE
F/us 11 calendar event display

### PR DESCRIPTION
**Pull request by KeeganWoods1**

### CODE REVIEW

File: src/app/routes/routes.page.ts [hunk 1]
```diff
@@ -44,11 +51,28 @@ export class RoutesPage implements OnInit, OnDestroy {
...
      */
-    private subscription: Subscription;
-
 
-    constructor(private formBuilder: FormBuilder, private routeFactory: RouteFactory) { }
+    private subscription: Subscription;
+    isFromCalendar: boolean;
+    eventFrom: string = 'H801';
+    eventTo: string;
+    //userLocation: string;
```
Comment (ClaudiaLapalme) [inline src/app/routes/routes.page.ts:59]: remove commented out line

File: src/app/routes/routes.page.ts [hunk 2]
```diff
@@ -44,11 +51,28 @@ export class RoutesPage implements OnInit, OnDestroy {
...
+    constructor(private formBuilder: FormBuilder, 
+                private routeFactory: RouteFactory, 
+                public activatedRoute: ActivatedRoute, 
+                public placesService: PlaceService,
+                public router: Router,
+                private route: ActivatedRoute) 
+                {
+                    this.route.queryParams.subscribe(params => {
+                        if (this.router.getCurrentNavigation().extras.state) {
+                          this.isFromCalendar = this.router.getCurrentNavigation().extras.state.isRouteToEvent
```
Comment (ClaudiaLapalme) [inline src/app/routes/routes.page.ts:70]: missing `;`

File: src/app/routes/routes.page.ts [hunk 3]
```diff
@@ -66,7 +90,15 @@ export class RoutesPage implements OnInit, OnDestroy {
                 if (this.form.valid) {
                     this.getRoutes();
                 }
-            });
+            }); 
+    }
+
+    ngAfterViewInit() {
+        if(this.isFromCalendar){
```
Comment (ClaudiaLapalme) [inline src/app/routes/routes.page.ts:97]: missing space before `{`

File: src/app/core/components/side-menu/side-menu.component.ts [hunk 4]
```diff
@@ -1,27 +1,60 @@
...
+        public calendarService: CalendarService,
+        public placeService: PlaceService,
+        public router: Router,
+        public locationService: LocationService,
+        public mapService: MapService,
+        public zone: NgZone
+    ) {}
 
     ngOnInit() {
+        this.locationService.getGeoposition().then(async (geoPos) => {
```
Comment (danseremet) [inline src/app/core/components/side-menu/side-menu.component.ts:43]: What is geoPos's type?

File: src/app/core/components/side-menu/side-menu.component.ts [hunk 5]
```diff
@@ -43,9 +76,20 @@ export class SideMenuComponent implements OnInit, OnDestroy {
...
-        this.userEmail = this.calendarService.getUserEmail();
-        document.getElementById('loggedInEmail').innerHTML = this.userEmail;
-        document.getElementById('loggedInPicture').innerHTML = '<img src=\"', this.userPicture, '\">';
+    setUserLocation(location: string): void {
+        this.userLocation = location;
     }
+
+    goToEvent() {
+        this.isRouteToEvent = true;
+        let navigationExtras: NavigationExtras = {
```
Comment (danseremet) [inline src/app/core/components/side-menu/side-menu.component.ts:85]: const

File: src/app/core/components/side-menu/side-menu.component.ts [hunk 6]
```diff
@@ -43,9 +76,20 @@ export class SideMenuComponent implements OnInit, OnDestroy {
...
+            state: {
+              location: this.calEventLocation,
+              userLocation: this.userLocation,
+              isRouteToEvent: this.isRouteToEvent
+            }
+          };
+          this.router.navigate(['routes'], navigationExtras);
```
Comment (danseremet) [inline src/app/core/components/side-menu/side-menu.component.ts:92]: alignment

File: src/app/core/services/calendar.service.ts [hunk 7]
```diff
@@ -6,22 +6,35 @@ import { AngularFireAuth } from '@angular/fire/auth';
...
-
+    fromCalendar: boolean = false;
 
     constructor(public afAuth: AngularFireAuth,
-        private platform: Platform,
-        private gplus: GooglePlus
-    ) { }
+                private platform: Platform,
+                private gplus: GooglePlus
+    ) { gapi.load('client'); }
```
Comment (danseremet) [inline src/app/core/services/calendar.service.ts:37]:
```suggestion
    ) { 
    	gapi.load('client'); 
    }
```

File: src/app/core/services/calendar.service.ts [hunk 8]
```diff
@@ -32,22 +45,79 @@ export class CalendarService {
...
+                  })
+                  .catch(error =>{
+                      console.log(error); //debug
+                  });
+            }).catch(error =>{
+                console.log(error); //debug
+            });
+        });    
+    }
+
```
Comment (danseremet) [inline src/app/core/services/calendar.service.ts:82]: There are extra whitespaces and missing spaces, run the formatter on this file.

File: src/app/core/services/calendar.service.ts [hunk 9]
```diff
@@ -56,23 +126,70 @@ export class CalendarService {
...
+                await gapi.client.calendar.events.list({
+                'calendarId': 'primary',
+                'timeMin': (new Date()).toISOString(),
+                'showDeleted': false,
+                'singleEvents': true,
+                'maxResults': 10,
+                'orderBy': 'startTime'
+              })
+              .then((response)=>{
+                const formattedTime = formatDate(response.result.items[0].start.dateTime, 'shortTime', 'en-US')
```
Comment (danseremet) [inline src/app/core/services/calendar.service.ts:151]: semi colon

File: src/app/routes/routes.page.ts [hunk 10]
```diff
@@ -44,11 +51,29 @@ export class RoutesPage implements OnInit, OnDestroy {
      * @private
      * @type {Subscription}
      */
-    private subscription: Subscription;
-
 
-    constructor(private formBuilder: FormBuilder, private routeFactory: RouteFactory) { }
+    private subscription: Subscription;
+    isFromCalendar: boolean;
+    eventFrom: string = 'H801';
```
Comment (ClaudiaLapalme) [inline src/app/routes/routes.page.ts:57]: can you make it start at the 8th floor escalator instead?

**Pull request merged**